File: G4G/RomanNumberToInt/solve.c
//{ Driver Code Starts
// Initial template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++

class Solution 
{
    map<char, int> values = 
    {
        {'I', 1},
        {'V', 5},
        {'X', 10},
        {'L', 50},
        {'C', 100},
        {'D', 500},
        {'M', 1000}
    };
    
  public:
    int romanToDecimal(string &str) 
    {
        int value = 0;
        
        for(int i = 0; i < str.length(); i++)
        {
            if(i < str.length() - 1)
            {
                if(values[str[i]] < values[str[i + 1]])
                {
                    value += values[str[i + 1]] - values[str[i]];
                    i++;
                }
                else
                    value += values[str[i]];
            }
            else
                value += values[str[i]];
        }
        
        return value;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;
        Solution ob;
        cout << ob.romanToDecimal(s) << endl;
    }
}
// } Driver Code Ends

File: G4G/SumofUpperAndLowerTris/solve.cs
//{ Driver Code Starts
//Initial Template for C#

using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {
                int n = Convert.ToInt32(Console.ReadLine());
                var stringArray = Console.ReadLine().Split(' ');
                List<List<int>> A = new List<List<int>>();
                for (int i = 0; i < n * n; i += n)
                {
                    List<int> row = new List<int>();
                    for (int j = 0; j < n; j++)
                    {
                        row.Add(int.Parse(stringArray[i + j]));
                    }
                    A.Add(row);
                }
                Solution obj = new Solution();
                var res = obj.sumTriangles(A, n);
                for (int i = 0; i < res.Count; i++)
                {
                    Console.Write(res[i] + " ");
                }
                Console.WriteLine();
                
                
            }

        }


    }

    
// } Driver Code Ends
//User function Template for C#

    class Solution
    {
        public int Sum(List<int> l, int start, int end)
        {
            int x = 0;
            for(int i = start; i < end; i++)
                x += l[i];
                
            return x;
        }
        
        //Function to return sum of upper and lower triangles of a matrix.
        public List<int> sumTriangles(List<List<int>> matrix, int n)
        {
            int u = 0;
            int l = 0;
            int uCount = 0;
            int lCount = 1;
            
            foreach(var a in matrix)
            {
                l += Sum(a, 0, lCount);
                u += Sum(a, uCount, n);
                uCount ++;
                lCount ++;
            }

            return new List<int>(){u, l};
        }

    }

}

//{ Driver Code Starts.
// } Driver Code Ends